packages/ripple-binary-codec/src/types/currency.ts

```diff
@@ -86,13 +88,14 @@ class Currency extends Hash160 {
   constructor(byteBuf: Buffer) {
     super(byteBuf ?? Currency.XRP.bytes)
     const code = this.bytes.slice(12, 15)
+    const hex = this.bytes.toString('hex')
 
-    if (this.bytes[0] !== 0) {
-      this._iso = null
-    } else if (/^0*$/.test(this.bytes.toString('hex'))) {
+    if (/^0*$/.test(hex)) {
```

Comment:
Should this check for exactly 40 zeros? (I believe currently this will match the empty string, and I don't know if that's correct)

Reply:
Ah, you're right. It does match for empty string. Maybe empty string also means to use XRP? @mvadari

Reply:
Seems handled elsewhere (Specifically bytesFromRepresentation)

Reply:
Empty string is an invalid currency code regardless. (We should only be allowing 3 character strings, or 40 character strings representing bytes. 0-2 character strings and 4-39 character strings are incorrectly formatted and should raise an exception)

Reply:
We can change this to check for a regex of 40 0s

Reply:
There could be an issue if you have a currency code of `"000"` (which is a valid currency code, but is not XRP)

Reply:
I changed it to requiring 40 0s.